TST,BLD,BUG: if __name__=='__main__': __main__.main(argv) and tests

westurner opened the pull request:
- fix asciifx console_script so that `pip -e .` works
- tests for main(argv=["--help"])

westurner committed:
TST,BLD,BUG: if __name__=='__main__': __main__.main(argv) and tests

fix asciifx console_script so that `pip -e .` works, tests for main(argv=["--help"])

westurner committed:
BUG,CLN,DOC: sys.exit(main(sys.argv[1:])), sort imports, module docstring

westurner commented:
I changed 'ascii-fx' to 'asciifx', because it already doesn't have a dash in pyproject.yml: 
https://github.com/apparebit/asciifx/pull/2/files#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711R37

apparebit commented:
Thank you for the pull request.

I just pushed an update that incorporates most of your changes.

Except, I omitted the tests because they strike me as overkill (live a little 🤪). Plus, I strongly dislike pytest, which performs far too much magic for a test runner. I'm surprised that you didn't add it to the optional dependencies.

I also added code to actually return meaningful *error* codes and not always 0.

If the changes meet your needs, I'll publish a new version of the package and close this pull request.

westurner commented:
Thanks! I had started to add syntax highlighting with pygments and added a
test main first, but haven't gotten around to adding a flag to keep syntax
highlighting optional. This was for an arithmetic tutorial for a science
fair -like STEM night, which I've continue to spend time on iterating.

There should be tests for agg in agg, which only produces .gif, which
*should* work into one longer video with e.g. ffmpeg and a screencast with
"GPU Screen Recorder" or OBS.

I have .gif output from agg that seems to flicker with --font-size=28 and
pygments-color-scheme=github-dark regardless of font.
The sphinxcontrib.asciicinema JS .cast player looks much more crisp; like
something that could be edited together with a screencast video of one or
more notebook implementations.

Perhaps CEF Chromium Embedded would be a way to replay notebooks in
terminal ASCII and record, and also to non-heaeless replay .ipynb input
cells in a browser just like asciifx.

I started to write an IPythonREPL(SubprocessRepl) that almost worked; but I
was in hurry due to a deadline so I punted for a PexpectRepl which also
almost worked. I don't think I had found the regex to strip terminal
control codes. Certain %magic commands like `%doctest_mode`, `%%pfile
asciiart.txt`, `%psource src.py`, `%pip` and `%mamba` only work with the
IPython interpreter, which also has shell escape codes to ignore when
pexpecting.

jupytext.read() and .reads() read any format of Jupyter Notebook; including
the `# %%` percent format.

IDK when I would've gotten to this again; probably in another year

On Sat, Feb 1, 2025, 6:45 PM Robert Grimm ***@***.***> wrote:

> Thank you for the pull request.
>
> I just pushed an update that incorporates most of your changes.
>
> Except, I omitted the tests because they strike me as overkill (live a
> little 🤪). Plus, I strongly dislike pytest, which performs far too much
> magic for a test runner. I'm surprised that you didn't add it to the
> optional dependencies.
>
> I also added code to actually return meaningful *error* codes and not
> always 0.
>
> If the changes meet your needs, I'll publish a new version of the package
> and close this pull request.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/apparebit/asciifx/pull/2#issuecomment-2629164454>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAMNS4RR5YL5EJRUY4OZCL2NVMBZAVCNFSM6AAAAABVYDHBC6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMMRZGE3DINBVGQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
